File: Python/Sort/sort.py
from array import array

def selectionSort(arr: array):
    for i in range(len(arr)-1):
        minIdx: int = i

        for j in range(i + 1, len(arr)):
            if (arr[minIdx] > arr[j]):
                minIdx = j

        temp: int = arr[i]
        arr[i] = arr[minIdx]
        arr[minIdx] = temp
        
    return arr

def insertionSort(arr: array):
    for i in range(1, len(arr)):
        currentIdx: int = i
        while currentIdx > 0 and arr[currentIdx-1] > arr[currentIdx]:
            temp: int = arr[currentIdx]
            arr[currentIdx] = arr[currentIdx-1]
            arr[currentIdx-1] = temp

            currentIdx -= 1

    return arr

File: JavaScript/Sort/sort.ts
export function selectionSort(arr: number[]): number[] {

    for (let i = 0; i <= arr.length - 2; i++) {
        let minIdx: number = i;

        for(let j = i + 1; j <= arr.length - 1; j++) {
            if (arr[minIdx] > arr[j]) {
                minIdx = j;
            }
        }

        let temp: number = arr[i];
        arr[i] = arr[minIdx];
        arr[minIdx] = temp;
    }
    return arr;
}

export function insertionSort(arr: number[]): number[] {
    for(let i = 1; i <= arr.length - 1; i++) {
        let currentIdx: number = i;
        while (currentIdx > 0 && arr[currentIdx - 1] > arr[currentIdx]) {
            let temp: number = arr[currentIdx];
            arr[currentIdx] = arr[currentIdx - 1];
            arr[currentIdx - 1] = temp;
            currentIdx--;
        }
    }
    
    return arr;
}